lib/index.js

```diff
@@ -128,9 +128,10 @@ function HttpProxyMiddleware(context, opts) {
     }
 
     function logError(err, req, res) {
-        var hostname = (req.hostname || req.host) || (req.headers && req.headers.host); // (node0.10 || node 4/5) || (websocket)
-        var targetUri = (proxyOptions.target.host || proxyOptions.target) + req.url;
+        var hostname = (req.headers && req.headers.host) || (req.hostname || req.host);     // (websocket) || (node0.10 || node 4/5)
+        var target = proxyOptions.target.host || proxyOptions.target;
+        var errReference = 'https://nodejs.org/api/errors.html#errors_common_system_errors'; // link to Node Common Systems Errors page
 
-        logger.error('[HPM] PROXY ERROR: %s. %s -> %s', err.code, hostname, targetUri);
+        logger.error('[HPM] Error occurred while trying to proxy request %s from %s to %s (%s) \n(%s)', req.url, hostname, target, err.code, errReference);
```

Comment:
Is the line-break really needed?
I can imagine when you configure `winston` in `logProvider` to log to a file; those line-breaks will add a lot of visual noise as you go through the logs...